lib/ui/text.dart

```diff
@@ -1501,6 +1545,40 @@ class ParagraphBuilder extends NativeFieldWrapperClass2 {
   }
   String _addText(String text) native 'ParagraphBuilder_addText';
 
+  /// Adds an inline placeholder space to the paragraph.
+  ///
+  /// The paragraph will contain a rectangular space with no text of the dimensions
+  /// specified.
+  ///
+  /// The [width] and [height] parameters specify the size of the rectangle, and
+  /// [baseline] indicates the distance of the alphabetic baseline from the top of
+  /// of the rectangle. This determines how the rectangle is aligned to the
+  /// surrounding text.
+  ///
+  /// For example, to make a 50x50 empty rectangle 'hole' that sits completely on
+  /// top of the alphabetic baseline, pass in `addPlaceholder(50, 50, 50)`. For
+  /// a box that is vertically centered on the alphabetic baseline, pass in
+  /// `addPlaceholder(50, 50, 25);`.
+  ///
+  /// Lines are permitted to break around each placeholder.
+  ///
+  /// Decorations will be drawn based on the font defined in the most recently
+  /// pushed TextStyle. Since placeholders are sized completely separately from
+  /// the rest of the text, the positions of the decorations are unlikely to
+  /// line up as expected. To hide or manually adjust decorations to fit, a text
+  /// style with the desired decoration behavior should be pushed before adding a
+  /// placeholder.
+  ///
+  /// Any decorations drawn through a placeholder will exist on the same canvas/layer
+  /// as the text. This means any content drawn on top of the space reserved by
+  /// the placeholder will be drawn over the decoration, possibly obscuring the
+  /// decoration.
+  void addPlaceholder(double width, double height, PlaceholderAlignment alignment, TextBaseline baseline, double baselineOffset) {
```

Comment:
baseline and baselineOffset should be named arguments (with baselineOffset having a default value), so that this is easier to use. (baseline in particular only needs to be set in rare cases)